core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaStoreReaderThread.java

```diff
@@ -106,7 +106,23 @@ public KafkaStoreReaderThread(String bootstrapBrokers,
 
     this.consumer = new KafkaConsumer<>(consumerProps);
 
-    List<PartitionInfo> partitions = this.consumer.partitionsFor(this.topic);
+    // Include a few retries since topic creation may take some time to propagate and schema registry is often started
+    // immediately after creating the schemas topic.
+    int retries = 0;
+    List<PartitionInfo> partitions = null;
+    while (retries < 3) {
+      partitions = this.consumer.partitionsFor(this.topic);
+      if (partitions != null && partitions.size() >= 1)
+        break;
+
+      try {
+        Thread.sleep(1000);
+      } catch (InterruptedException e) {
+        // ignore
+      }
+      retries++;
```

Comment:
i'd include the increment in the while condition. Or, if here i'd write retries += 1, which I think is semantically slightly better.